lib/page-object.rb

```diff
@@ -53,19 +54,20 @@ module PageObject
   # a method named initialize_accessors if it exists. Upon initialization of 
   # the page it will call a method named initialize_page if it exists.
   #
-  # @param [Watir::Browser or Selenium::WebDriver::Driver] the platform browser to use
+  # @param [Watir::Browser, Watir::HTMLElement or Selenium::WebDriver::Driver, Selenium::WebDriver::Element] the platform browser/element to use
   # @param [bool] open the page if page_url is set
   #
-  def initialize(browser, visit=false)
+  def initialize(root, visit=false)
     initialize_accessors if respond_to?(:initialize_accessors)
-    initialize_browser(browser)
```

Comment:
This is the meat of the change. The page object initialization takes a "root" which can be an element or a browser. The browser is gathered from there, and the root element is set if an element is given.